### ISSUE TITLE
Error: "can't modify frozen Hash"

### ISSUE THREAD

**Issue by etobia** [ahx/openapi_first#276]
I'm using the example from [the docs](https://github.com/ahx/openapi_first/tree/main?tab=readme-ov-file#test-assertions) to register my OAS document for use in minitest. 

However, when I call `register` I get a `FrozenError` on [this line](https://github.com/ahx/openapi_first/blob/main/lib/openapi_first/test.rb#L14), presumably because the `DEFINITIONS` hash is frozen?

```console
gems/openapi_first-2.0.2/lib/openapi_first/test.rb:14:in `register': can't modify frozen Hash: {} (FrozenError)
```

If I patch the `OpenapiFirst::Test` module the subsequent call to `register` works, but I wonder if I'm missing something in my setup. 

```ruby
  module OpenapiFirst
    module Test
      class << self
        alias original_definitions definitions

        def definitions
          @mutable_definitions ||= original_definitions.dup
        end
      end
    end
  end

```

Thoughts?

**Comment by ahx** [comment]
Hi @etobia . Thanks for the report. I don’t think your setup is wrong. The „freeze“ must have been added by my misconfigured linter by accident.
And apparently this part is missing a test.
I am on a trip right now, but I have hotfixed this on main. Can you see if works for you now?

**Comment by etobia** [comment]
Thanks for the quick response @ahx.  I just tried `8c46ba7` and it appears to be working!

**Comment by ahx** [comment]
Great to hear, @etobia. I am on a trip until July 15th and unfortunately I won’t be able to release a new version until then. But I will once I am back.

**Comment by etobia** [comment]
No worries at all. Thanks again!

**Comment by ahx** [comment]
Just released the fix for this in 2.0.3
